src/Tools/IdeCoreBenchmarks/IncrementalSourceGeneratorBenchmarks.cs

```diff
@@ -0,0 +1,191 @@
+        {
+            RestoreCompilerSolution();
+            SetUpWorkspace();
+        }
+
+        [IterationSetup]
+        public void IterationSetup() => LoadSolutionAsync().Wait();
+
+        private void RestoreCompilerSolution()
+        {
+            var roslynRoot = Environment.GetEnvironmentVariable(Program.RoslynRootPathEnvVariableName);
+            _solutionPath = Path.Combine(roslynRoot, @"Compilers.sln");
+            var restoreOperation = Process.Start("dotnet", $"restore /p:UseSharedCompilation=false /p:BuildInParallel=false /m:1 /p:Deterministic=true /p:Optimize=true {_solutionPath}");
+            restoreOperation.WaitForExit();
+            if (restoreOperation.ExitCode != 0)
+                throw new ArgumentException($"Unable to restore {_solutionPath}");
+        }
+
+        private static void SetUpWorkspace()
+        {
+            // QueryVisualStudioInstances returns Visual Studio installations on .NET Framework, and .NET Core SDK
+            // installations on .NET Core. We use the one with the most recent version.
+            var msBuildInstance = MSBuildLocator.QueryVisualStudioInstances().OrderByDescending(x => x.Version).First();
+
+            MSBuildLocator.RegisterInstance(msBuildInstance);
+        }
+
+        private async Task LoadSolutionAsync()
+        {
+            var roslynRoot = Environment.GetEnvironmentVariable(Program.RoslynRootPathEnvVariableName);
+            _solutionPath = Path.Combine(roslynRoot, @"Roslyn.sln");
+
+            if (!File.Exists(_solutionPath))
+                throw new ArgumentException("Couldn't find Roslyn.sln");
+
+            Console.WriteLine("Found Roslyn.sln: " + Process.GetCurrentProcess().Id);
+            var assemblies = MSBuildMefHostServices.DefaultAssemblies
+                .Add(typeof(AnalyzerRunnerHelper).Assembly)
+                .Add(typeof(FindReferencesBenchmarks).Assembly);
+            var services = MefHostServices.Create(assemblies);
+
+            _workspace = MSBuildWorkspace.Create(new Dictionary<string, string>
+                {
+                    // Use the latest language version to force the full set of available analyzers to run on the project.
+                    { "LangVersion", "9.0" },
+                }, services);
+
+            if (_workspace == null)
+                throw new ArgumentException("Couldn't create workspace");
+
+            Console.WriteLine("Opening roslyn.  Attach to: " + Process.GetCurrentProcess().Id);
+
+            var start = DateTime.Now;
+
+            var solution = _workspace.OpenSolutionAsync(_solutionPath, progress: null, CancellationToken.None).Result;
+            Console.WriteLine("Finished opening roslyn: " + (DateTime.Now - start));
+        }
+
+        [IterationCleanup]
+        public void IterationCleanup()
+        {
+            _workspace.Dispose();
+            _workspace = null;
+        }
+
+        [Benchmark]
+        public async Task RunGeneratorOld()
```

Comment:
>RunGeneratorOld [](http://example.com/codeflow?start=26&length=15)

Should this be named "RunGenerator" instead?

Reply:
This is just personal benchmark code.  I view it as a playground, vs a distinct section that must follow particular rules.  :)   i can make changes, but it's just not a critical thing from my perspective.

Reply:
done.